plugins/autodim.koplugin/main.lua

```diff
@@ -204,6 +207,19 @@ function AutoDim:onSuspend()
     end
 end
 
+function AutoDim:updateFooter(clear)
+    -- update footer only if it is not covered by another widget
+    if self.top_widget_before_dim == "ReaderUI" or
+        (self.top_widget_before_dim ~= "ConfigDialog" and self.top_widget_before_dim ~= "Screensaver"
```

Comment:
I'm not entirely sure ConfigDialog can ever be a top widget, but I may be remembering this wrong ;).

Reply:
Yes it can, when opening the bottom menu and wait for AutoDim knocks in.

Reply:
Just a thought: Should we add an attribute `covering_footer` to the identified widgets (ReaderUI, ConfigDialog, ScreenSaver, VirtualKeyboard) to make things more compact and move the check to `onUpdateFooter` in `readerfooter.lua`?

Reply:
Probably not, too niche ;).

Reply:
I was thinking of something else last night, my bad ;p.

I still think it's not necessarily worth the hassle, but am less strongly opposed to it than I was last night ;p.

Reply:
(IIRC, there's also the cases where we want onUpdateFooter to update the data without necessarily refreshing the screen).

Reply:
Maybe let us keep that idea in the back of our heads for now and implement it when another pice of code has similar requirements.

Reply:
My thoughts exactly ;).